examples/platform/silabs/provision/ProvisionStorageDefault.cpp

```diff
@@ -70,16 +70,26 @@ CHIP_ERROR ErasePage(uint32_t addr)
     return chip::DeviceLayer::Silabs::GetPlatform().FlashErasePage(addr);
 }
 
-CHIP_ERROR WritePage(uint32_t addr, const uint8_t * data, size_t size)
+size_t RoundNearest(size_t n, size_t multiple)
 {
-    return chip::DeviceLayer::Silabs::GetPlatform().FlashWritePage(addr, data, size);
+    return (n % multiple) > 0 ? n + (multiple - n % multiple) : n;
 }
 
-size_t RoundNearest(size_t n, size_t multiple)
+CHIP_ERROR WritePage(uint32_t addr, const uint8_t * data, size_t size, size_t max)
 {
-    return (n % multiple) > 0 ? n + (multiple - n % multiple) : n;
+    size_t size_32 = RoundNearest(size, 4);
+    if(size_32 != size)
+    {
+        uint8_t * p = (uint8_t *) data;
+        VerifyOrReturnError(size_32 <= max, CHIP_ERROR_BUFFER_TOO_SMALL);
```

Comment:
So essentially this error means that the buffer is smaller than a page. Seems like it would be more explicit to check that from the start, since this method is Called _WritePage_

Reply:
Suggestion applied.